### PR TITLE
bpo-43369: sqlite3_column_text() and sqlite3_column_blob() failures now raise MemoryError

### PR DESCRIPTION
**Pull request by erlend-aasland**
https://bugs.python.org/issue43369

### CODE REVIEW

File: Lib/sqlite3/test/types.py [hunk 1]
```diff
@@ -237,6 +237,15 @@ def test_number2(self):
         # if the converter is not used, it's an int instead of a float
         self.assertEqual(type(value), float)
 
+    def test_convert_zero_sized_blob(self):
+        self.con.execute("create table b(b cblob)")
+        sqlite.converters["CBLOB"] = lambda x: b'blobish'
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/types.py]: This and line 246 can be moved into `setUp` and `tearDown` methods respectively.

File: Lib/sqlite3/test/types.py [hunk 2]
```diff
@@ -237,6 +237,15 @@ def test_number2(self):
...
+        self.con.execute("insert into b(b) values (?)", (b'',))
```
Comment (berkerpeksag) [inline Lib/sqlite3/test/types.py]: Nitpick: Also, please use either single or double quotes consistently in new code.

File: Modules/_sqlite/cursor.c [hunk 3]
```diff
@@ -280,17 +281,19 @@ _pysqlite_fetch_one_row(pysqlite_Cursor* self)
          * See https://sqlite.org/c3ref/column_blob.html for details.
          */
         if (converter != Py_None) {
-            const char *blob = (const char*)sqlite3_column_blob(self->statement->st, i);
+            const void *blob = sqlite3_column_blob(self->statement->st, i);
```
Comment (berkerpeksag) [inline Modules/_sqlite/cursor.c:284]: There is also this one in line 343:

```c
const char *blob = sqlite3_column_blob(self->statement->st, i);
````

**Review by berkerpeksag — CHANGES_REQUESTED** [review]

**Comment by erlend-aasland** [comment]
Thank you for reviewing, @berkerpeksag. I have made the requested changes; please review again.

**Review by berkerpeksag — APPROVED** [review]
LGTM! Hopefully we aren't going to introduce another regression this time :)

**Comment by erlend-aasland** [comment]
> LGTM! Hopefully we aren't going to introduce another regression this time :)

Let's hope so :) Thanks, Berker!

**Pull request merged**